src/main/java/com/linecorp/armeria/common/reactivestreams/RichPublisher.java

```diff
@@ -24,6 +24,14 @@
 
 public interface RichPublisher<T> extends Publisher<T> {
     boolean isOpen();
+
+    /**
+     * Returns {@code true} if this stream has been closed and did not publish any elements.
+     * Note this method will not return {@code true} when the stream is open even if it did not
```

Comment:
Not that this method will not...

...it has not published anything...

...because it may publish something later.